docs/backends.md

```diff
@@ -705,3 +705,66 @@ type: Opaque
 data:
   password: <path:secret-id#key | base64encode>
 ```
+
+### Kubernetes Secret
+
+Setup the backend with default secret from which to conveniently inject values from.
+Optionally alternative secret names can be passed through path.
+
+**Note**: The Kubernetes Secret backend does not support versioning
+
+##### Kubernetes Secret Authentication
+
+Backend inherits same in-cluster service-account as the plugin itself
+
+These are the parameters for Kubernetes Secret:
+
+```
+AVP_TYPE: kubernetessecret
+AVP_K8S_SECRET_SECRET: avp-secret-data
+```
+`AVP_K8S_SECRET_SECRET` can have namespace prefix, ie.:
+```
+AVS_K8S_SECRET_SECRET: prod:avp-secret-data
+```
+
+##### Examples
+
+###### Inline Path Reading from configured secret
+
+```yaml
+kind: Secret
+apiVersion: v1
+metadata:
+  name: test-secret
+type: Opaque
+data:
+  username: <path:~#key>
+  password: <path:default#key2>
+```
+
+###### Path Annotation
+
+```yaml
+kind: Secret
+apiVersion: v1
+metadata:
+  name: test-secret
+  annotations:
+    avp.kubernetes.io/path: "custom-secret-data"
+type: Opaque
+data:
+  password: <key>
+```
+
+###### Inline Path
+
+```yaml
+kind: Secret
+apiVersion: v1
+metadata:
+  name: test-secret
+type: Opaque
+data:
+  password: <path:custom-secret-data#key>
```

Comment:
Can you add an example for an inline path pulling a secret from another namespace?